dockstore-webservice/src/main/java/io/dockstore/webservice/core/language_parsing/LanguageParsingRequest.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2021 OICR and UCSC
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.dockstore.webservice.core.language_parsing;
+
+import io.dockstore.common.DescriptorLanguage;
+import io.dockstore.webservice.core.SourceFile;
+import io.swagger.v3.oas.annotations.media.Schema;
+import java.util.List;
+import java.util.Objects;
+
+@Schema(description = "Request sent to the external language parsing service")
+public class LanguageParsingRequest {
+
+    @Schema(description = "The Git URI", required = true)
+    private String uri;
+
+    @Schema(description = "The Git branch/tag", required = true)
+    private String branch;
```

Comment:
the branch is interesting, could any problems arise if a branch is rapidly changing or does it not matter since the last change might win anyway? alternatively, should this be a commit?

Reply:
Our versions table uses branch/tag and not commit id, so this is consistent with our current functionality